tests/app/views/test_brief_response.py

```diff
@@ -607,6 +738,103 @@ def test_can_not_submit_an_invalid_brief_response(
             }
         }
 
+    def test_can_not_submit_brief_response_with_non_legacy_data(self, live_dos_framework):
+        # To create a brief_resonse with an invalid key from the new schema, we need to switch the feature flag on.
+        # We switch it back off after creation to allow us to exhibit the behaviour we're testing.
+        feature_flag_date = (datetime.utcnow() - timedelta(days=7)).strftime("%Y-%m-%d")
+        self.app.config["FEATURE_FLAGS_NEW_SUPPLIER_FLOW"] = feature_flag_date
+
+        non_legacy_brief_response_data = {
+            'essentialRequirementsMet': True,
+            'niceToHaveRequirements': [True, False, True, False, True],
+            'availability': u'a',
+            'respondToEmailAddress': 'supplier@email.com'
+        }
+
+        create_res = self.create_brief_response(data=non_legacy_brief_response_data)
+        assert create_res.status_code == 201
+
+        brief_response_id = json.loads(create_res.get_data(as_text=True))['briefResponses']['id']
+
+        self.app.config["FEATURE_FLAGS_NEW_SUPPLIER_FLOW"] = self.feature_flag_date
+
+        submit_res = self._submit_brief_response(brief_response_id)
+        data = json.loads(submit_res.get_data(as_text=True))
+
+        assert submit_res.status_code == 400
+        assert "'essentialRequirementsMet' was unexpected" in data['error']['_form'][0]
+        assert data['error']['essentialRequirements'] == "answer_required"
+
+
+class TestSubmitBriefReponseWhenFeatureFlagIsOff(TestSubmitBriefReponseForBriefCreatedBeforeFeatureFlag):
+
+    feature_flag_date = False
+
+    def setup(self):
+        super(TestSubmitBriefReponseWhenFeatureFlagIsOff, self).setup()
+        self.app.config["FEATURE_FLAGS_NEW_SUPPLIER_FLOW"] = self.feature_flag_date
```

Comment:
Can we have a comment here explaining what the flag is being set to?

I'd quite like it for all the other times we set it tbh, so it's clear what's the flag is being set to.

An alternative is having two attributes on one of the base classes, something like `date_one_week_ago`, `date_one_week_in_the_future`. Would mean we could write things like:

`self.app.config["FEATURE_FLAGS_NEW_SUPPLIER_FLOW"] = self.date_one_week_in_the_future`